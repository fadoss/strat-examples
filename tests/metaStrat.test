metalevel/metaStrat

red metaSrewrite(upModule('TEST, false), upTerm("deleveled"), repeat-until('palin[none]{empty}, 'C:Char, nil), breadthFirst, 0) .
red metaSrewrite(upModule('TEST, false), upTerm("deleveled"), repeat-until('palin[none]{empty}, 'C:Char, nil), breadthFirst, 1) .
red metaSrewrite(addRepeat(upModule('TEST, false), 'strip, 'palin[none]{empty}), upTerm("evitative"), 'strip[[upTerm(3)]], breadthFirst, 0) .
red metaSrewrite(addRepeat(upModule('TEST, false), 'strip, 'palin[none]{empty}), upTerm("evitative"), 'strip[[upTerm(3)]], breadthFirst, 1) .
red metaSrewrite(upModule('TEST, false), upTerm(bigPalindrome(100)), ('palin[none]{empty}) !, breadthFirst, 0) .
red metaSrewrite(upModule('TEST, false), upTerm(bigPalindrome(100)), toBasis(('palin[none]{empty}) !), breadthFirst, 0) .
red metaSrewrite(upModule('TEST, false), upTerm(bigPalindrome(100)), toBasis(('palin[none]{empty}) !), depthFirst, 0) .
