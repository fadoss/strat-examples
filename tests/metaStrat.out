==========================================
reduce in TEST : metaSrewrite(upModule('TEST, false), upTerm("deleveled"),
    repeat-until('palin[none]{empty}, 'C:Char, nil), breadthFirst, 0) .
rewrites: 40
result ResultPair: {'"v".Char, 'Char}
==========================================
reduce in TEST : metaSrewrite(upModule('TEST, false), upTerm("deleveled"),
    repeat-until('palin[none]{empty}, 'C:Char, nil), breadthFirst, 1) .
rewrites: 4
result ResultPair?: (failure).ResultPair?
==========================================
reduce in TEST : metaSrewrite(addRepeat(upModule('TEST, false), 'strip, 'palin[
    none]{empty}), upTerm("evitative"), 'strip[[upTerm(3)]], breadthFirst, 0) .
rewrites: 32
result ResultPair: {'"tat".String, 'String}
==========================================
reduce in TEST : metaSrewrite(addRepeat(upModule('TEST, false), 'strip, 'palin[
    none]{empty}), upTerm("evitative"), 'strip[[upTerm(3)]], breadthFirst, 1) .
rewrites: 5
result ResultPair?: (failure).ResultPair?
==========================================
reduce in TEST : metaSrewrite(upModule('TEST, false), upTerm(bigPalindrome(
    100)), ('palin[none]{empty}) !, breadthFirst, 0) .
rewrites: 656
result ResultPair: {'"".String, 'String}
==========================================
reduce in TEST : metaSrewrite(upModule('TEST, false), upTerm(bigPalindrome(
    100)), toBasis(('palin[none]{empty}) !), breadthFirst, 0) .
rewrites: 1134
result ResultPair: {'"".String, 'String}
==========================================
reduce in TEST : metaSrewrite(upModule('TEST, false), upTerm(bigPalindrome(
    100)), toBasis(('palin[none]{empty}) !), depthFirst, 0) .
rewrites: 1132
result ResultPair: {'"".String, 'String}
Bye.
